tt_metal/distributed/mesh_device.cpp

```diff
@@ -160,16 +163,41 @@ size_t SystemMesh::Impl::get_num_devices() const {
     return num_rows * num_cols;
 }
 
+chip_id_t SystemMesh::Impl::get_physical_device_id(size_t logical_row_idx, size_t logical_col_idx) const {
+    TT_FATAL(
+        logical_row_idx < logical_mesh_shape_.num_rows,
+        "Row index out of bounds: {} >= {}",
+        logical_row_idx,
+        logical_mesh_shape_.num_rows);
+    TT_FATAL(
+        logical_col_idx < logical_mesh_shape_.num_cols,
+        "Column index out of bounds: {} >= {}",
+        logical_col_idx,
+        logical_mesh_shape_.num_cols);
+    auto logical_coordinate = Coordinate{logical_row_idx, logical_col_idx};
+    auto physical_coordinate = logical_to_physical_coordinates_.at(logical_coordinate);
+    return physical_coordinate_to_device_id_.at(physical_coordinate);
```

Comment:
Several places look up physical coordinates in the first map, then device_id in the second map... Does it make sense to store the mapping from logical to device_id?

Reply:
Yeah, that's sensible -- I've added it in.